pkg/peermgr/swarm.go

```diff
@@ -136,35 +136,42 @@ func (swarm *Swarm) Start() error {
 	}
 
 	for id, addr := range swarm.multiAddrs {
-		go func(id uint64, addr *peer.AddrInfo) {
+		go func(id uint64, addr *peer.AddrInfo, ctx context.Context) {
 			if err := retry.Retry(func(attempt uint) error {
-				// for restart node, after updating the routing table, some nodes may not exist in routing table
-				routers := swarm.notifiee.getPeers()
-				if _, ok := routers[id]; !ok {
-					swarm.logger.Infof("Can't find node %d from routing table, stopping connect", id)
+				select {
+				case <- swarm.ctx.Done():
+					swarm.cancel()
```

Comment:
cancel() should have  been called before to trigger ctx.Done()

Reply:
remove it